server/src/services/hcs-ballot-processing.service.ts

```diff
@@ -96,38 +94,53 @@ export class HcsBallotProcessingService {
 			this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Invalid Voting Completion Time.`);
 		} else if (ballot.startTimestamp >= ballot.endTimestamp) {
 			this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Voting ending time preceeds starting.`);
-		} else if (computeDiffInDays(ballot.startTimestamp, ballot.endTimestamp) < this.config.minVotingThreshold) {
-			this.logger.verbose(
-				`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Voting ending time does not permit required voting windo of ${this.config.minVotingThreshold} days.`,
-			);
 		} else if (ballot.consensusTimestamp > ballot.startTimestamp) {
 			this.logger.verbose(
 				`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Proposal Voting Starting Time preceeds Proposal Creation Time.`,
 			);
-		} else if (computeDiffInDays(ballot.consensusTimestamp, ballot.startTimestamp) < this.config.minimumStandoffPeriod) {
-			this.logger.verbose(
-				`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Voting starting time is less than ${this.config.minimumStandoffPeriod} days from ballot creation.`,
-			);
 		} else if (isNaN(ballot.minVotingThreshold) || ballot.minVotingThreshold < 0 || ballot.minVotingThreshold > 1) {
 			this.logger.verbose(
 				`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Invalid Minimum Voting Threshold, must be a fraction between 0 and 1.0 inclusive.`,
 			);
-		} else if (ballot.minVotingThreshold < this.config.minVotingThreshold) {
-			this.logger.verbose(
-				`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Invalid Minimum Voting Threshold, must be equal to or greater than global configuration of ${this.config.minVotingThreshold}.`,
-			);
 		} else if (!Array.isArray(ballot.ineligibleAccounts)) {
 			this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Ineligible account list is not an array.`);
 		} else if (ballot.ineligibleAccounts.find((a) => !is_entity_id(a))) {
 			this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Invalid account id found in Ineligible account list.`);
 		} else if (-1 !== ballot.ineligibleAccounts.findIndex((a) => !is_entity_id(a))) {
 			this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Found invalid account id in list of inelegible accounts.`);
 		} else {
-			ballot.ineligibleAccounts = [...new Set([...ballot.ineligibleAccounts, ...this.config.ineligibleAccounts])];
 			return async () => {
-				if (await this.dataService.getBallot(ballot.consensusTimestamp)) {
+				const rule = this.dataService.getLatestRule(hcsMirrorRecord.consensus_timestamp as unknown as TimestampKeyString);
```

Comment:
All rule related validation needs to be moved into this async function. This guarantees that all previous valid messages are in the data service. i.e. If a Rule was submitted 1 sequence before this ballot, then that rule would be used for this ballot.